contracts/EuroPool.sol

```diff
@@ -0,0 +1,327 @@
+  IERC20 public usdc;
+  uint24 public usdcEuroPoolFee;
+
+  uint256 public numOrders;
+
+  uint256 constant rsaModulusChunksLen = 17;
+  uint256[rsaModulusChunksLen] public revolutServerKeys = [
+    1357877033487705755044473051962315017,
+    1992913992454963774321360338650810838,
+    1332679464913641539987869420620959603,
+    2348132427872155825755866901316180901,
+    1883906444029850560136606276757266463,
+    2450997574449119024683227690782146111,
+    2287069363878927698261632102663638321,
+    2446813570435318449337400488662901236,
+    48497691420903457,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0
+  ];
+
+  constructor(
+    IPool _usdcPool,
+    IRevolutSendVerifier _verifier,
+    IERC20 _ageuro,
+    ISwapRouter _swapRouter,
+    IQuoterV2 _quoter,
+    uint24 euroUsdcPoolFee,
+    IERC20 _usdc
+  ) {
+    usdcPool = _usdcPool;
+    verifier = _verifier;
+    ageuro = _ageuro;
+    swapRouter = _swapRouter;
+    quoter = _quoter;
+    usdcEuroPoolFee = euroUsdcPoolFee;
+    usdc = _usdc;
+
+    // Set initial orderId
+    numOrders = 1;
+  }
+
+  // Update helpers
+  function updateVerifier(IRevolutSendVerifier _verifier) public {
+    verifier = _verifier;
+  }
+
+  function updateUsdcPool(IPool _usdcPool) public {
+    usdcPool = _usdcPool;
+  }
+
+  function updateUSDC(IERC20 _usdc) public {
+    usdc = _usdc;
+  }
+
+  function updateSwapRouter(ISwapRouter _swapRouter) public {
+    swapRouter = _swapRouter;
+  }
+
+  function updateEuro(IERC20 _ageuro) public {
+    ageuro = _ageuro;
+  }
+
+  function updateQuoter(IQuoterV2 _quoter) public {
+    quoter = _quoter;
+  }
+
+  function updateEuroUsdcPoolFee(uint24 _usdcEuroPoolFee) public {
+    usdcEuroPoolFee = _usdcEuroPoolFee;
+  }
+
+  // Approve helpers
+  function approveUsdcToUniswap() public {
+    usdc.approve(address(swapRouter), type(uint256).max);
+  }
+
+  // Pool functions
+
+  // Called by the LP
+  // Locks agEURO in the pool
+  // Creates an order in the pool
+
+  // Note: Before create order make sure to approve the contract to spend the amount of agEURO you want to lock.
+
+  function createOrder(
+    uint256 amount,
+    string memory offChainPaymentAddress
+  ) external override {
+    orders[numOrders] = Order({
+      id: numOrders,
+      amount: amount,
+      offChainPaymentAddress: offChainPaymentAddress,
+      claimed: false,
+      completedAt: 0,
+      takerEmailHash: [uint256(0), uint256(0), uint256(0)]
+    });
+
+    ageuro.transferFrom(msg.sender, address(this), amount);
+
+    numOrders += 1;
+  }
+
+  // Called by the taker
+  // Claims an order in the pool
+  function claimOrder(
+    uint256 id,
+    bytes32 emailHash // will be set to 0 for the demo
+  ) external override {
+    require(!orders[id].claimed, "Order has already been claimed.");
+    claimedOrders[id] = emailHash;
+    orders[id].claimed = true;
+  }
+
+  // Called by the taker
+  // Completes an order in the pool
+  function completeOrder(
```

Comment:
For our demo, do we even need the completeOrder method? We just need the off-ramp method.